File: Competitive Programming 3/Section 1.3.3/11683 - Laser Sculpture/Python/main.py
def main():
    while True:
        try:
            A,C=map(int,input().split())
        except:
            return
        if A==0 and C == 0:
            return
        last,ans=A,0
        for current in map(int,input().split()):
            if current > last:
                ans += current - last
            last = current
        print(ans+A-last)
main()

File: Competitive Programming 3/Section 1.3.3/11683 - Laser Sculpture/C++/main.cpp
#include <iostream>

using namespace std;

int main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    int A,C;
    while(cin>>A,A){
        cin >> C;
        int last=A,ans=0,current;
        for(int i = 0 ; i < C ; i ++){
            cin >> current;
            if(current>last){
                ans+=current-last;
            }
            last = current;
        }
        cout << ans+A-last << "\n";
    }
    return 0;
}